scripts/Package.hs

```diff
@@ -0,0 +1,133 @@
+findProgram whereToSearch name = do
+    fmap fst <$> findProgramOnSearchPath silent whereToSearch name
+
+download :: String -> FilePath -> IO ()
+download url destPath = callProcess "curl" ["-fSL", "-o", destPath, url]
+
+find7z :: IO (Maybe FilePath)
+find7z = do
+    let default7zPath = "C:\\Program Files\\7-Zip"
+    findProgram [ProgramSearchPathDefault, ProgramSearchPathDir default7zPath] "7z"
+
+get7zPath :: IO FilePath
+get7zPath = find7z >>= \case
+    Just programPath -> return programPath
+    Nothing          -> error errorMsg
+    where errorMsg = "cannot find 7z, please install from https://7-zip.org.pl/ or make sure that program is visible in PATH"
+
+unpack7z :: FilePath -> FilePath -> IO ()
+unpack7z archive outputDirectory = do
+    programPath <- get7zPath
+    callProcess programPath ["x", "-y", "-o" <> outputDirectory, archive]
+
+pack7z :: [FilePath] -> FilePath -> IO ()
+pack7z packedPaths outputArchivePath = do
+    programPath <- get7zPath
+    callProcess programPath $ ["a", "-y", outputArchivePath] <> packedPaths
+
+buildWithMsBuild :: FilePath -> IO ()
+buildWithMsBuild solutionPath = do
+    let msbuildPath = "C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Community\\MSBuild\\15.0\\Bin\\amd64\\MSBuild.exe"
+    callProcess msbuildPath ["/property:Configuration=Release", solutionPath]
+
+copyToDir :: FilePath -> FilePath -> IO ()
+copyToDir destDir sourcePath = do
+    createDirectoryIfMissing True destDir
+    putStrLn $ "Copy " ++ sourcePath ++ " to " ++ destDir
+    let destPath = destDir </> takeFileName sourcePath
+    copyFile sourcePath destPath
+
+pushArtifact :: FilePath -> IO ()
+pushArtifact path = do
+    callProcess "appveyor" ["PushArtifact", path]
+
+-- Function downloads 7z to temp folder, so it doesn't leave any trash behind.
+downloadAndUnpack7z :: FilePath -> FilePath -> IO ()
+downloadAndUnpack7z archiveUrl targetDirectory = do
+    withSystemTempDirectory "" $ \tmpDir -> do
+        let archiveLocalPath = tmpDir </> takeFileName archiveUrl
+        download archiveUrl archiveLocalPath
+        unpack7z archiveLocalPath targetDirectory
+
+-- We need to extract the package with dev libraries and set the environment
+-- variable DATAFRAMES_DEPS_DIR so the MSBuild project recognizes it.
+--
+-- The package contains all dependencies except for Python (with numpy).
+-- Python needs to be provided by CI environment and pointed to by `PythonDir`
+-- environemt variable.
+prepareEnvironment :: FilePath -> IO ()
+prepareEnvironment tempDir = do
+    let depsDirLocal = tempDir </> "deps"
+    downloadAndUnpack7z depsArchiveUrl depsDirLocal
+    setEnv "DATAFRAMES_DEPS_DIR" depsDirLocal
+
+-- Copies subdirectory with all its contents between two directories
+copyDirectory :: FilePath -> FilePath -> FilePath -> IO ()
+copyDirectory sourceDirectory targetDirectory subdirectoryFilename = do
+    let from = sourceDirectory </> subdirectoryFilename
+    let to = targetDirectory </> subdirectoryFilename
```

Comment:
unnecessary let

Reply:
Do we have a policy for that? I'm not sure what is the consensus among Haskell community, but I find writing lets increase readability of the code and can be handy when refactoring, e.g. the lines can be moved without editing.
In C++ it is usually considered a bad practice to introduce multiple variable names with a single type declaration specifier.

Reply:
I agree. As far as I know it has also different meaning, you can't do:
```
let x = y + 1
let y = 4
```
I think policy in Haskell is: less words = better :D